src/foam/nanos/http/FileServlet.java

```diff
@@ -53,28 +52,41 @@ private void fileNotFoundError(HttpServletResponse resp, String file) {
   protected void doGet(HttpServletRequest req, HttpServletResponse resp) {
     String   pathInfo = req.getPathInfo();
     String   filePath = pathInfo.substring(SERVLET_NAME.length() + 2);
-    String   fileName = "";
-    String[] paths    = filePath.split("/");
-
-    if( paths.length > 0 ) {
-      fileName = paths[paths.length - 1];
-    } else {
-      fileNotFoundError(resp, filePath);
-      return;
-    }
-
-    String[] tokens = fileName.split("\\.");
-    String extension = tokens.length > 0 ? tokens[tokens.length-1 ] : "unknown";
 
     try {
-      File   srcFile = new File(filePath);
+      File   srcFile = new File(filePath.isEmpty() ? "./" : filePath);
       String path    = srcFile.getAbsolutePath();
       String cwd     = System.getProperty("user.dir");
 
-      if ( srcFile.isFile() && srcFile.canRead() && cwd.equals(path.substring(0, cwd.length())) ) {
+      if ( srcFile.isDirectory() && srcFile.canRead() && cwd.equals(path.substring(0, cwd.length())) ) {
+        resp.setContentType(extLookup.get("html"));
+        PrintWriter pw = resp.getWriter();
+        pw.write(
+            "<!DOCTYPE html>\n" +
+                "<html>\n" +
+                "<body>\n" +
+                "<ul style=\"list-style-type:disc\">");
+
+        File[] files = srcFile.listFiles();
+        if(files != null && files.length > 0) {
```

Comment:
Follow the style-guide.

https://github.com/foam-framework/foam/wiki/FOAM2-Style-Guidelines